app/src/keymap.c

```diff
@@ -1,24 +1,32 @@
 
+#include <sys/util.h>
 #include <logging/log.h>
 LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/matrix.h>
 #include <zmk/keymap.h>
 #include <dt-bindings/zmk/matrix-transform.h>
-#include <sys/util.h>
+#include <drivers/behavior.h>
+#include <zmk/behavior.h>
 
 static u32_t zmk_keymap_layer_state = 0;
 static u8_t zmk_keymap_layer_default = 0;
 
-#if DT_NODE_HAS_PROP(ZMK_KEYMAP_NODE, transform)
-#define ZMK_KEYMAP_TRANSFORM_NODE DT_PHANDLE(ZMK_KEYMAP_NODE, transform)
-#define ZMK_KEYMAP_LEN DT_PROP_LEN(ZMK_KEYMAP_TRANSFORM_NODE, map)
+#define ZMK_KEYMAP_NODE DT_CHOSEN(zmk_keymap)
+#define ZMK_KEYMAP_LAYERS_LEN DT_PROP_LEN(ZMK_KEYMAP_NODE, layers)
+
+#define LAYER_NODE(l) DT_PHANDLE_BY_IDX(ZMK_KEYMAP_NODE, layers, l)
 
-#define _TRANSFORM_ENTRY(i, l) \
```

Comment:
This all got simplified when the matrix transforms were moved *lower*, so the keymap *only* operates on key positions now, not on rows/columns at all.